<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1130</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1130</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1130</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Balance car monitor[www.dancells.com]</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>logo.ico</normaloff>logo.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(255, 255, 255);</string>
  </property>
  <widget class="AXBOpenGLWidget" name="openGLWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>340</y>
     <width>311</width>
     <height>221</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_Send">
   <property name="geometry">
    <rect>
     <x>1040</x>
     <y>570</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton{
border: 2px solid rgb(255, 129, 25);
border-radius: 5px;
color: rgb(255, 129, 25);}
QPushButton:hover{
color: rgb(255, 255, 255);
background-color: rgb(255, 129, 25);
}
</string>
   </property>
   <property name="text">
    <string>发送</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="textEdit_SBuff">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>470</y>
     <width>701</width>
     <height>121</height>
    </rect>
   </property>
  </widget>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>0</y>
     <width>791</width>
     <height>431</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>1</number>
   </property>
   <widget class="QWidget" name="tab">
    <attribute name="title">
     <string>接收文本</string>
    </attribute>
    <widget class="QTextEdit" name="textEdit_RBuff">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>761</width>
       <height>391</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_2">
    <attribute name="title">
     <string>曲线绘制</string>
    </attribute>
    <widget class="QCustomPlot" name="customPlot" native="true">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>701</width>
       <height>391</height>
      </rect>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>720</x>
       <y>10</y>
       <width>52</width>
       <height>141</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <item row="0" column="0">
       <widget class="QCheckBox" name="checkBox_ch0">
        <property name="text">
         <string>CH0</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_CH</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="checkBox_ch1">
        <property name="text">
         <string>CH1</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="tristate">
         <bool>false</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_CH</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="checkBox_ch2">
        <property name="text">
         <string>CH2</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_CH</string>
        </attribute>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="checkBox_ch3">
        <property name="text">
         <string>CH3</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_CH</string>
        </attribute>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="checkBox_ch4">
        <property name="text">
         <string>CH4</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_CH</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QTabWidget" name="tabWidget_2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>311</width>
     <height>291</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="tab_3">
    <attribute name="title">
     <string>参数设置</string>
    </attribute>
    <widget class="QWidget" name="layoutWidget1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>291</width>
       <height>51</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="horizontalSpacing">
       <number>0</number>
      </property>
      <item row="1" column="1">
       <widget class="QRadioButton" name="radioBtn_Ges">
        <property name="text">
         <string>姿态</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_Mode</string>
        </attribute>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>运行模式:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QRadioButton" name="radioBtn_Spd">
        <property name="text">
         <string>速度</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_Mode</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="radioBtn_Ctrl">
        <property name="text">
         <string>控制</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_Mode</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QRadioButton" name="radioBtn_Erect">
        <property name="text">
         <string>直立</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_Mode</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QRadioButton" name="radioBtn_Turn">
        <property name="text">
         <string>转向</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_Mode</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>71</y>
       <width>291</width>
       <height>192</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout" columnstretch="1,2,1,2,1,2,2">
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label_G">
        <property name="text">
         <string>姿态参数：</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5" colspan="2">
       <widget class="QRadioButton" name="ratioBtn_isSave">
        <property name="text">
         <string>是否保存</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_GKp">
        <property name="text">
         <string>Kp:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="lineEdit_GKp"/>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_GKi">
        <property name="text">
         <string>Ki:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLineEdit" name="lineEdit_GKi"/>
      </item>
      <item row="1" column="6">
       <widget class="QPushButton" name="pushButton_G">
        <property name="styleSheet">
         <string notr="true">QPushButton{
border: 2px solid rgb(255, 129, 25);
border-radius: 5px;
color: rgb(255, 129, 25);}
QPushButton:hover{
color: rgb(255, 255, 255);
background-color: rgb(255, 129, 25);
}
</string>
        </property>
        <property name="text">
         <string>写入</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QLabel" name="label_S">
        <property name="text">
         <string>速度环：</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_SKp">
        <property name="text">
         <string>Kp:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="lineEdit_SKp"/>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="label_SKi">
        <property name="text">
         <string>Ki:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QLineEdit" name="lineEdit_SKi"/>
      </item>
      <item row="3" column="4">
       <widget class="QLabel" name="label_SKd">
        <property name="text">
         <string>Kd:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="5">
       <widget class="QLineEdit" name="lineEdit_SKd"/>
      </item>
      <item row="3" column="6">
       <widget class="QPushButton" name="pushButton_S">
        <property name="styleSheet">
         <string notr="true">QPushButton{
border: 2px solid rgb(255, 129, 25);
border-radius: 5px;
color: rgb(255, 129, 25);}
QPushButton:hover{
color: rgb(255, 255, 255);
background-color: rgb(255, 129, 25);
}
</string>
        </property>
        <property name="text">
         <string>写入</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QLabel" name="label_E">
        <property name="text">
         <string>直立环：</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_EKp">
        <property name="text">
         <string>Kp:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="lineEdit_EKp"/>
      </item>
      <item row="5" column="2">
       <widget class="QLabel" name="label_EKi">
        <property name="text">
         <string>Ki:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QLineEdit" name="lineEdit_EKi"/>
      </item>
      <item row="5" column="4">
       <widget class="QLabel" name="label_EKd">
        <property name="text">
         <string>Kd:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="5">
       <widget class="QLineEdit" name="lineEdit_EKd"/>
      </item>
      <item row="5" column="6">
       <widget class="QPushButton" name="pushButton_E">
        <property name="styleSheet">
         <string notr="true">QPushButton{
border: 2px solid rgb(255, 129, 25);
border-radius: 5px;
color: rgb(255, 129, 25);}
QPushButton:hover{
color: rgb(255, 255, 255);
background-color: rgb(255, 129, 25);
}
</string>
        </property>
        <property name="text">
         <string>写入</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QLabel" name="label_T">
        <property name="text">
         <string>转向环：</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_TKp">
        <property name="text">
         <string>Kp:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLineEdit" name="lineEdit_TKp"/>
      </item>
      <item row="7" column="2">
       <widget class="QLabel" name="label_TKi">
        <property name="text">
         <string>Ki:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="3">
       <widget class="QLineEdit" name="lineEdit_TKi"/>
      </item>
      <item row="7" column="4">
       <widget class="QLabel" name="label_TKd">
        <property name="text">
         <string>Kd:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="5">
       <widget class="QLineEdit" name="lineEdit_TKd"/>
      </item>
      <item row="7" column="6">
       <widget class="QPushButton" name="pushButton_T">
        <property name="styleSheet">
         <string notr="true">QPushButton{
border: 2px solid rgb(255, 129, 25);
border-radius: 5px;
color: rgb(255, 129, 25);}
QPushButton:hover{
color: rgb(255, 255, 255);
background-color: rgb(255, 129, 25);
}
</string>
        </property>
        <property name="text">
         <string>写入</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_4">
    <attribute name="title">
     <string>小车控制</string>
    </attribute>
    <widget class="QWidget" name="layoutWidget3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>337</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="2">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>速度:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="label_speed">
        <property name="text">
         <string>10 m/s</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QProgressBar" name="progressBar_eq">
        <property name="value">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>电量:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_angle">
        <property name="text">
         <string>55.0</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>倾角:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="rocker" name="rocker_Ctrl">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>50</y>
       <width>200</width>
       <height>200</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>440</y>
     <width>112</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>发送缓冲区:</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget4">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>570</y>
     <width>311</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,3,1,3,1,3">
    <item row="0" column="0">
     <widget class="QLabel" name="label_Pitch">
      <property name="text">
       <string>Pitch:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="label_Pitch_v">
      <property name="styleSheet">
       <string notr="true">QLabel {color: rgb(255,0,0);}</string>
      </property>
      <property name="text">
       <string>180.00</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="label_Roll">
      <property name="text">
       <string>Roll:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QLabel" name="label_Roll_v">
      <property name="styleSheet">
       <string notr="true">QLabel {color:rgb(0,255,0);}</string>
      </property>
      <property name="text">
       <string>180.00</string>
      </property>
     </widget>
    </item>
    <item row="0" column="4">
     <widget class="QLabel" name="label_Yaw">
      <property name="text">
       <string>Yaw:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="5">
     <widget class="QLabel" name="label_Yaw_v">
      <property name="styleSheet">
       <string notr="true">QLabel {color:rgb(0,0,255);}</string>
      </property>
      <property name="text">
       <string>180.00</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget5">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>0</y>
     <width>291</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_5" columnstretch="1,4,1">
    <item row="0" column="0">
     <widget class="QLabel" name="label_Addr">
      <property name="text">
       <string>IP:Port:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="label_Addr_v">
      <property name="text">
       <string>255.255.255.255:8080</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="pushButton_Connect">
      <property name="styleSheet">
       <string notr="true">QPushButton{
border: 2px solid rgb(255, 129, 25);
border-radius: 5px;
color: rgb(255, 129, 25);}
QPushButton:hover{
color: rgb(255, 255, 255);
background-color: rgb(255, 129, 25);
}
</string>
      </property>
      <property name="text">
       <string>连接</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AXBOpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">axbopenglwidget.h</header>
  </customwidget>
  <customwidget>
   <class>rocker</class>
   <extends>QGraphicsView</extends>
   <header>rocker.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup_CH">
   <property name="exclusive">
    <bool>false</bool>
   </property>
  </buttongroup>
  <buttongroup name="buttonGroup_Mode"/>
 </buttongroups>
</ui>
